kf_lib_data_ingest/etl/stage_analyses.py

```diff
@@ -80,5 +80,35 @@ def check_counts(discovery_sources, expected_counts):
     return passed, '\n'.join(message)
 
 
-def compare_counts(discovery_sources_one, discovery_sources_two):
-    False, "Compare Counts NYI"
+def compare_counts(
+    name_one, discovery_sources_one, name_two, discovery_sources_two
+):
+    message = [f'COMPARING COUNTS BETWEEN {name_one} and {name_two}']
+    passed = True
+
+    setA = set(discovery_sources_one.keys())
+    setB = set(discovery_sources_two.keys())
+    if not (setA == setB):
+        message.append('')
+        message.append(f'Column keys not equal')
+        message.append(f'{name_one} {setA}')
+        message.append('vs')
+        message.append(f'{name_two} {setB}')
+        message.append('Difference is:')
+        message.append(setA ^ setB)
+        passed = False
```

Comment:
Do you think this should always result in an error? Will there ever be a scenario where someone chooses not to merge all of their extract dfs into the single df?

Reply:
I don't know.

Reply:
Well the other option is just that this doesn't result in a fail. Idk, I could go either way here

Reply:
I'd rather set this to fail now and try to find alternate ways of dealing with your proposed scenario.

Reply:
👍